src/components/LmsConfigurations/CanvasIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,275 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import isEmpty from 'lodash/isEmpty';
+import {
+  ValidationFormGroup, Input, StatefulButton, Icon,
+} from '@edx/paragon';
+import { snakeCaseFormData } from '../../utils';
+import StatusAlert from '../StatusAlert';
+import LmsApiService from '../../data/services/LmsApiService';
+import NewRelicService from '../../data/services/NewRelicService';
+import SUBMIT_STATES from '../../data/constants/formSubmissions';
+
+export const REQUIRED_CONFIG_FIELDS = [
+  'clientId',
+  'clientSecret',
+  'canvasAccountId',
+  'canvasBaseUrl',
+];
+
+class CanvasIntegrationConfigForm extends React.Component {
+  state = {
+    invalidFields: {},
+    submitState: SUBMIT_STATES.DEFAULT,
+    active: this.props.config?.active,
+    error: undefined,
+  };
+
+  createCanvasConfig = async (formData) => {
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      await LmsApiService.postNewCanvasConfig(transformedData);
+      this.setState({
+        error: undefined,
+      });
+      return undefined;
+    } catch (error) {
+      return this.handleErrors(error);
+    }
+  }
+
+  updateCanvasConfig = async (formData, uuid) => {
+    const transformedData = snakeCaseFormData(formData);
+    try {
+      await LmsApiService.updateCanvasConfig(transformedData, uuid);
+      this.setState({
+        error: undefined,
+      });
+      return undefined;
+    } catch (error) {
+      return this.handleErrors(error);
+    }
+  }
+
+  validateCanvasConfigForm = (formData, requiredFields) => {
+    const invalidFields = requiredFields
+      .filter(field => !formData.get(field))
+      .reduce((prevFields, currField) => ({ ...prevFields, [currField]: true }), {});
+    return invalidFields;
+  }
+
+  handleSubmit = async (formData, config) => {
+    await this.setState({ submitState: SUBMIT_STATES.PENDING });
+    const invalidFields = this.validateCanvasConfigForm(formData, REQUIRED_CONFIG_FIELDS);
+    if (!isEmpty(invalidFields)) {
+      this.setState({
+        invalidFields: {
+          ...invalidFields,
+        },
+        submitState: SUBMIT_STATES.DEFAULT,
+      });
+      return;
+    }
+
+    formData.append('enterprise_customer', this.props.enterpriseId);
+    formData.set('active', this.state.active);
```

Comment:
I think I said some useless things in my previous comment.
What I was trying to understand is if you create a config, don't refresh, and then submit again, does the formData *only* include fields changed or does it just pass everything along again? Are the fields passed along based on the data actually entered in or does it trigger the props again (and thus everything resets?)

I think if formData only passes along fields modified, then you don't need to do anything with this active field. Only in the case where a user actually changes that flag would it need to be passed along and in that case it would be as part of formData.

However, if formData just passes everything along and uses whatever values were in the props, then yeah not only am I concerned about this field, but about every field, because there wouldn't be any props on initial creation.

If I'm sounding confusing (which is very likely) we can chat more offline. I want to clarify I'm not dictating a change here, I'm actually just unsure what formData is passing in this case. I've found it passes everything, but you're using an onChange override that updates state you don't otherwise seem to be using, so I'm not sure if formData gets updated in the same way....